### PR TITLE
Bring back AsyncCommands

### PR DESCRIPTION
**Pull request by marius-se**
Brings async commands back by adding a new property `asyncCommands` to `Application`.

### CODE REVIEW

**Comment by marius-se** [comment]
I just saw my formatter went for a little rampage in `Application.swift`, I hope that's not a problem :D

File: Sources/Vapor/Application.swift [hunk 1]
```diff
@@ -163,18 +161,31 @@ public final class Application: Sendable {
...
+        let combinedCommands = asyncCommands.merge(
+            with: commands,
+            defaultCommand: commands.defaultCommand,
+            help: commands.help
+        )
+
         var context = CommandContext(console: self.console, input: self.environment.commandInput)
         context.application = self
-        try self.console.run(command, with: context)
+        try self.eventLoopGroup.next()
```
Comment (0xTim) [inline Sources/Vapor/Application.swift:184]:
```suggestion
        try self.eventLoopGroup.any()
```

File: Sources/Vapor/Core/Core.swift [hunk 2]
```diff
@@ -77,6 +83,7 @@ extension Application {
                 var commands = Commands()
                 commands.use(BootCommand(), as: "boot")
                 self.commands = .init(commands)
+                self.asyncCommands = .init(AsyncCommands())
```
Comment (0xTim) [inline Sources/Vapor/Core/Core.swift:86]: Tempted to say we should switch all the internals over but I think that can wait

**Review by 0xTim — CHANGES_REQUESTED** [review]
One nit and we need to update both package manifests to point to 4.13.0

File: Sources/Vapor/Core/Core.swift (hunk 2 above)
Comment (marius-se) [reply Sources/Vapor/Core/Core.swift:86]: Yeah I was also thinking about that, but I'd say that belongs to a separate PR. 👍🏼

File: Tests/AsyncTests/AsyncCommandsTests.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            let help = "Does the foo."
+
+            func run(using context: CommandContext, signature: Signature) throws {
+                context.console.output("Hello, \(signature.name)!")
+            }
+        }
+
+        app.asyncCommands.use(FooCommand(), as: "foo")
+
+        app.environment.arguments = ["vapor", "foo", "bar"]
```
Comment (0xTim) [inline Tests/AsyncTests/AsyncCommandsTests.swift:24]: Shouldn't this be
```suggestion
        app.environment.arguments = ["vapor", "foo", "--name", "bar"]
```

File: Tests/AsyncTests/AsyncCommandsTests.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+        XCTAssertNoThrow(try app.start())
```
Comment (0xTim) [inline Tests/AsyncTests/AsyncCommandsTests.swift:26]: Is there any way we can verify it runs, like incrementing a global counter? A manual run will be good enough if you haven't done that. (Do we have any tests for normal commands)

File: Tests/AsyncTests/AsyncCommandsTests.swift (hunk 3 above)
Comment (marius-se) [reply Tests/AsyncTests/AsyncCommandsTests.swift:24]: No it's an `@Argument` not `@Option`

File: Tests/AsyncTests/AsyncCommandsTests.swift (hunk 4 above)
Comment (marius-se) [reply Tests/AsyncTests/AsyncCommandsTests.swift:13]: Good idea, I just pushed a commit. Unfortunately we don't have any tests for commands (or I couldn't find one)

**Review by 0xTim — APPROVED** [review]
🥳

**Pull request merged**